app/src/main/java/ru/vssemikoz/newsfeed/MainActivity.java

```diff
@@ -44,38 +47,79 @@ protected void onCreate(Bundle savedInstanceState) {
         KEY = mainApplication.getKEY();
 
         ImageButton categoryButton = findViewById(R.id.ib_category);
+        ImageButton favoriteNewsButton = findViewById(R.id.ib_favorite);
         categoryButton.setOnClickListener(v -> {
             DialogFragment categoryDialog = new PickCategoryDialog();
             categoryDialog.show(getSupportFragmentManager(), "categoryDialog");
         });
+        favoriteNewsButton.setOnClickListener(v -> {
+            favoriteNewsState = !favoriteNewsState;
+            Log.d(TAG, "onCreate: " + favoriteNewsState);
+            newsItemsFromDB = getNewsFromDB();
+            changeFavoriteIcon(favoriteNewsButton);
+            adapter.setNewsList(newsItemsFromDB);
+            adapter.notifyDataSetChanged();
+        });
+        updateCategoryNameOnToolBar();
         initRecView();
         initNewsItemDAO();
         initNewsItemListCallback();
         performCall();
     }
 
+    private void changeFavoriteIcon(ImageButton button) {
+        if (favoriteNewsState){
+            button.setImageResource(R.drawable.ic_star_yellow_48dp);
+        }else {
+            button.setImageResource(R.drawable.ic_star_white_48dp);
+        }
+    }
+
+    private List<NewsItem> getNewsFromDB() {
+        if (favoriteNewsState){
+            if (category == Category.ALL){
+                return newsItemDAO.getFavoriteNews();
+            }else{
+                return newsItemDAO.getFavoriteNewsByCategory(category.name());
+            }
+        }else{
+            if (category == Category.ALL){
+                return newsItemDAO.getAll();
+            }else{
+                return newsItemDAO.getNewsByCategory(category.name());
+            }
+        }
+    }
+
     private void initRecView(){
         recyclerView =  findViewById(R.id.rv_news_feed);
         recyclerView.setLayoutManager(new LinearLayoutManager(this));
+        adapter = new NewsFeedAdapter(getApplicationContext());
+        adapter.setOnItemClickListener(this::changeFollowState);
+    }
+
+    private void changeFollowState(int position) {
```

Comment:
Why have we called this method `changeFollowState`? I don't see any `Follow` functionality there.